pkg/resolution/resolver/git/resolver.go

```diff
@@ -154,12 +155,14 @@ func (r *Resolver) resolveAPIGit(ctx context.Context, params map[string]string)
 	if content == nil || len(content.Data) == 0 {
 		return nil, fmt.Errorf("no content for resource in %s/%s %s", params[orgParam], params[repoParam], params[pathParam])
 	}
+
 	return &resolvedGitResource{
 		Content:  content.Data,
 		Revision: content.Sha,
 		Org:      params[orgParam],
 		Repo:     params[repoParam],
 		Path:     content.Path,
+		URL:      fmt.Sprintf("%s%s/%s.git", scmClient.BaseURL.String(), params[orgParam], params[repoParam]),
```

Comment:
> Be very careful with this, since I'm not sure if this will work with every URL type - e.g. will this match the URL if I give it ssh:// or git://? You may want to just pass through the user provided value.

In https://github.com/tektoncd/pipeline/pull/5450, provider api was introduced to allow users to only provide `repo` and `org` instead of a complete `url`. So what I am trying to do here is to construct the real url based on user input `repo` and `org` because the url is needed in the provenance configSource.

Reply:
I do have some doubts here about what is the right way to get the real url using the provider api. In current implementation, the url for github example is `https://api.github.com/<org>/<repo>.git` if users use Authenticated API
approach by providing `org` and `repo`.

@abayer do you have any idea or better suggestion here? Thanks

Reply:
What do we intend to use the value of the URL annotation for? Is it supposed to be the clone URL, or a web link? Or just a consistent identifier of the source of the resolved resource? If it's meant to be the clone URL or a web link, then we'd probably want to add 
```
repo, _, err := scmClient.Repositories.Get(ctx, fmt.Sprintf("%s/%s", params[orgParam], params[repoParam]"))
if err != nil { ... }

...

return &resolvedGitResource{
...
  URL: repo.Clone, // or repo.Link if we want the web link
}
```

If it's just meant to be a consistent identifier, then adding a new annotation like `resolution.tekton.dev/server` and using `serverURL` for that with nothing set for `resolution.tekton.dev/url` when using the API-based resolution feels appropriate.

Reply:
Thanks @abayer !! very helpful!
The intention is for SLSA provenance to record the [identifier of the config source](https://slsa.dev/provenance/v0.2#invocation.configSource.uri). I think `repo.Link` should be what is desired here.